packages/web-app/src/hooks/useDaoDetailsQuery.tsx

```diff
@@ -0,0 +1,114 @@
+import {Client, DaoDetails} from '@aragon/sdk-client';
+import {useQuery} from '@tanstack/react-query';
+import {useCallback, useEffect} from 'react';
+import {useNavigate, useParams} from 'react-router-dom';
+
+import {useNetwork} from 'context/network';
+import {resolveDaoAvatarIpfsCid} from 'utils/library';
+import {NotFound} from 'utils/paths';
+import {useClient} from './useClient';
+
+/**
+ * Fetches DAO data for a given DAO address or ENS name using a given client.
+ * @param client - The client to use for the request.
+ * @param daoAddressOrEns - The DAO address or ENS name to fetch data for.
+ * @returns A Promise that resolves to the DAO data.
+ * @throws An error if the client is not defined or if the DAO address/ENS name is not defined.
+ */
+async function fetchDaoDetails(
+  client: Client | undefined,
+  daoAddressOrEns: string | undefined
+): Promise<DaoDetails | null> {
+  if (!daoAddressOrEns)
+    return Promise.reject(new Error('daoAddressOrEns must be defined'));
+
+  if (!client) return Promise.reject(new Error('client must be defined'));
+
+  const daoDetails = await client.methods.getDao(daoAddressOrEns.toLowerCase());
+  return daoDetails;
+}
+
+/**
+ * Custom hook to fetch DAO details for a given DAO address or ENS name using the current network and client.
+ * @param daoAddressOrEns - The DAO address or ENS name to fetch details for.
+ * @returns An object with the status of the query and the DAO details, if available.
+ */
+export const useDaoQuery = (
+  daoAddressOrEns: string | undefined,
+  refetchInterval = 0
+) => {
+  const {network, networkUrlSegment} = useNetwork();
+  const {client, network: clientNetwork} = useClient();
+
+  // if network is unsupported this will be caught when compared to client
+  const queryNetwork = networkUrlSegment ?? network;
+
+  // make sure that the network and the url match up with client network before making the request
+  const enabled =
+    !!daoAddressOrEns && !!client && clientNetwork === queryNetwork;
+
+  const queryFn = useCallback(() => {
+    return fetchDaoDetails(client, daoAddressOrEns);
+  }, [client, daoAddressOrEns]);
+
+  return useQuery<DaoDetails | null>({
+    queryKey: ['daoDetails', daoAddressOrEns, queryNetwork],
+    queryFn,
+    select: addAvatarToDao,
+    enabled,
+    refetchOnWindowFocus: false,
+    refetchInterval: () => (refetchInterval ? refetchInterval : false),
```

Comment:
```suggestion
    refetchInterval: () => refetchInterval || false,
```

gpt suggested this not me xD

Reply:
Actually I think I can just pass in `refetchInterval` if It's defaulted to 0 because it knows not to refetch when the value is 0